include/fmt/core.h

```diff
@@ -2520,7 +2540,17 @@ FMT_CONSTEXPR auto parse_format_specs(ParseContext& ctx)
       mapped_type_constant<T, context>::value != type::custom_type,
       decltype(arg_mapper<context>().map(std::declval<const T&>())),
       typename strip_named_arg<T>::type>;
+#if defined(__cpp_if_constexpr)
+  if constexpr (std::is_default_constructible_v<
+                    formatter<mapped_type, char_type>>) {
+    return formatter<mapped_type, char_type>().parse(ctx);
+  } else {
+    type_is_unformattable_for<T, char_type> _;
+    return ctx.begin();
+  }
+#else
```

Comment:
Why do we need to duplicate the logic here?

Reply:
So I could do something like this instead:
```cpp
if constexpr (/* not default constructible */) {
    type_is_unformattable_for<T, char_type> _;
}
return formatter<mapped_type, char_type>().parse(Ctx);
```
But doing it this way gives you both the ~good~ slightly better error message with `type_is_unformattable` and also the "not default constructible" error message when you actually try to construct the thing. But all the latter one tells you is that... something failed, whereas the former tells you _what_ failed. I think it's better to just provide the one?

Reply:
I am not sure why we care about checking default constructibility at all. It should be uncommon and not worth any extra diagnostics.